### PR TITLE
Also add ref to non-sortable branch in TableHeader

### LINKED ISSUE #4876: Forward refs for TableHeader <th>
### Summary

In developing column resizing for the data table for Watson IoT PAL, we're unable to set width on `<th>` elements. We need the ref to these elements to avoid imperatively setting the width from event callbacks (`event.target.closest('th')` for example). 

`TableHeader` is a function component so we're unable to use the `ref` attribute on it. We're unable to wrap these with a container without breaking table semantics, and the styling for table headers as well.

### Justification

Multiple Watson IoT products require the use of resizable columns.

### Specific timeline issues / requests

This work is in progress already. I'd like to get this in ASAP to avoid reimplementing `TableHeader` ourselves.

### Available extra resources

I can submit a PR

### PR DESCRIPTION
**Pull request by gmoehler**
Closes #4876 (well actually it is already closed we are still adding to it ;-) , maybe should re-open it?)
https://github.com/carbon-design-system/carbon/pull/4877 added forward refs to the `TableHeader` but did forget to add the ref to one codebranch. This PR is fixing this issue.

@tay1orjones 
@asudoh

### CODE REVIEW

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @gmoehler!

**Comment by asudoh** [comment]
BTW could you fix the code format? Thanks!

File: packages/react/src/components/DataTable/TableHeader.js [hunk 1]
```diff
@@ -64,7 +64,7 @@ const TableHeader = React.forwardRef(function TableHeader(
 ) {
   if (!isSortable) {
     return (
-      <th {...rest} className={headerClassName} scope={scope} colSpan={colSpan}>
+      <th {...rest} className={headerClassName} scope={scope} colSpan={colSpan}> ref={ref}
```
Comment (joshblack) [inline packages/react/src/components/DataTable/TableHeader.js]:
```suggestion
      <th {...rest} className={headerClassName} scope={scope} colSpan={colSpan} ref={ref}>
```

**Review by joshblack — CHANGES_REQUESTED** [review]

**Comment by gmoehler** [comment]
@asudoh The react deployment probably needs a re-start

**Review by joshblack — APPROVED** [review]

**Review by tw15egan — APPROVED** [review]

**Pull request merged**